pkg/daemon/ceph/client/command.go

```diff
@@ -36,62 +34,41 @@ const (
 	cmdExecuteTimeout = 1 * time.Minute
 )
 
-func FinalizeCephCommandArgs(command string, args []string, configDir, clusterName string) (string, []string) {
-	// If the command should be run inside the toolbox pod, include the kubectl args to call the toolbox
-	if RunAllCephCommandsInToolbox {
-		toolArgs := []string{"-it", "exec", "rook-ceph-tools", "-n", clusterName, "--", command}
-		return Kubectl, append(toolArgs, args...)
-	}
-
-	// No need to append the args if it's the default ceph cluster
-	if clusterName == "ceph" && configDir == "/etc" {
-		return command, args
-	}
-
-	// Append the args to find the config and keyring
-	confFile := fmt.Sprintf("%s.config", clusterName)
-	keyringFile := fmt.Sprintf("%s.keyring", AdminUsername)
-	configArgs := []string{
```

Comment:
Hmm ... It's my intention that this PR simplifies Rook a little by assuming that these locations are consistent for all daemons, eliminating the need to keep track of where the configs are since it's always static. Inside the pods/containers (but not necessarily outside of the containers), all of those values are left at default, meaning the cluster is always assumed to be named `ceph`, and therefore the keyring is at `/etc/ceph/keyring`, and the config is at `/etc/ceph/ceph.conf`. And because they are in `/etc/ceph`, they are guaranteed to always persist from the config-init container to the daemon container, and they are currently guaranteed not to be persisted to disk, which I believe we do not want since they are generated each time a daemon is started.

If some part of this doesn't sound quite right, I may have misunderstood or made an assumption I shouldn't have.

Reply:
Agreed on the intention of the changes. That works perfectly for the daemon containers, which will never have to worry about any cluster but their own.

The exception to this pattern is the operator, which may need to manage multiple clusters. The operator can't rely on the default paths such as `/etc/ceph` because that only works when the tools have a single cluster to manage. The operator will therefore need to use configs in different directories. Hopefully scoping these changes to the daemons won't complicate things too much.

Reply:
Thanks for clarifying this. I definitely have to make sure I consider this. I believe there is an integration test that tests multiple clusters, yes? Would you think this would detect any errors I might accidentally introduce in the operator?

Reply:
Yes, I would hope the integration test(s) will catch any issues around this, especially the one that runs two clusters from the same operator. I'll plan on taking this for a spin as well when you're getting closer.